amlogic: changes for qca9377 sdio module

chewitt opened the pull request:
The QCA9377 module on the SML-5442TW box requires a large patch to bluez that has been refused upstream as it uses hciattach; the patch author is still working up the enthusiasm to rework it around a new/preferred method. The module also requires ath10k-sdio firmware that doesn't currently exist in the mainline kernel-firmware repo so we add a temporary package to source it, along with a package for the BT firmware. In the long-term I expect to drop or roll both packages into others, but it's been a good way to experiment with different approaches for being more explicit with what gets included, or in this case excluded to avoid bloat and select only what's actually needed for the device.

kszaq commented:
Why not have it in master so that 3.14 devices would also benefit? There are at least a few boxes I can think of that use QCA9377 driver. If this is something that shouldn't be used for all projects, create a patch directory to be used by AML projects. I guess that there might be some benefit for Generic if there is any QCA9377-based BT.

chewitt commented:
As noted in the original PR comment, both this and ath10k things are temp packages and will probably roll into something else at some point. Once we have a broader view on what the firmware needs are for AML boxes on mainline we'll restructure things and rebase/drop stuff as required.

kszaq commented:
Sorry for not being clear - there already is a driver and firmware package for 3.14 devices using QCA9377 but there is no `bluez` patch in master that would allow using BT. It would be nice to have the `bluez` patch applied for 3.14 devices so that people can use their BT devices. It's been included in the community builds for quite a long time with no negative effect.

chewitt commented:
I dropped the firmware packages from the PR (will submit those to amlogic separately) and moved the bluez patches to the normal location. The patch set needs to be retested on 3.14 because I have additional patches from https://github.com/boundarydevices/android_external_bluez/commits/bluez-5.47-qca that are not in the single (older) patch currently being added to community builds.

chewitt commented:
I've finally tested this on 3.14 kernel now and things seem to be fine.
